Library/Client.cs

```diff
@@ -203,6 +203,10 @@ public HttpStatusCode PerformRequest(HttpRequestMethod method, string urlPath)
                         errors = Error.ReadResponseAndParseErrors(response);
                         throw new InvalidCredentialsException(errors);
 
+                    case HttpStatusCode.BadRequest:

```

Comment:
[I believe](https://msdn.microsoft.com/en-us/library/06tc147t.aspx) you can move this case label atop `case HttpStatusCode.PreconditionFailed:` on line 210 to since they throw the same exception.